### PR TITLE
Sharing public posts view without logging in

### PR DESCRIPTION
**Pull request by mxiuwang**
* Public posts (including unlisted public posts)'s links can be copied and shared without the user having to log in.
* [Fixed]: Copying links to friends/inbox posts now works, and the post can only be viewed if you log in and are following the author.

If this is good, I will close the other PR which limits the scope of the copy link and share buttons.

### CODE REVIEW

File: code/socialDistribution/views.py [hunk 1]
```diff
@@ -919,3 +913,20 @@ def inbox(request):
...
     return render(request, 'author/inbox.html', context)
+
+def public_share(request, id):
+    """
+        Renders a single public post shared via link (including unlisted public posts)
+    """
+    post = get_object_or_404(LocalPost, pk=id) 
+    author = post.author
+   
+    if not post.is_public:
```
Comment (debecerra) [inline code/socialDistribution/views.py:924]: Friends posts are currently being called. `post.is_public` is a method so it needs to be called.

File: code/socialDistribution/models/post.py [hunk 2]
```diff
@@ -261,7 +266,10 @@ def get_id(self):
         return f"{API_BASE}/author/{self.author.id}/posts/{self.id}"
     
     def get_local_shareable_link(self):
-        return f"{CLIENT_BASE}/posts/local/{self.id}"
+        try:
+            return f"{CLIENT_BASE}/public-share/{self.id}" # publicly accessible (doesn't have to login)
+        except:
+            return f"{CLIENT_BASE}/posts/local/{self.id}" # only accessible to logged in users
```
Comment (debecerra) [inline code/socialDistribution/models/post.py:272]: I'm not sure what you are trying to do here, could you please explain? The try clause will never throw any errors that need to be caught.

**Review by debecerra — CHANGES_REQUESTED** [review]
I'm not sure how this addresses the bug in the other PR but I do think this resolves #36 once a few issues are fixed. 

Also noticed that the "Copy Link" button does not work for inbox posts, but I'm not sure that it needs to. We only need to have a link for public posts created by our authors, so we could just hide this button for inbox posts. Any thoughts?

I'll let @tylerelston  or @JawadHossain address comments relating to the other PR.

File: code/socialDistribution/views.py (hunk 1 above)
Comment (mxiuwang) [reply code/socialDistribution/views.py]: thanks, fixed

File: code/socialDistribution/models/post.py (hunk 2 above)
Comment (mxiuwang) [reply code/socialDistribution/models/post.py]: it's the link that is only accessible to logged in users, but i got rid of it just to make things easier

**Comment by mxiuwang** [comment]
> I'm not sure how this addresses the bug in the other PR but I do think this resolves #36 once a few issues are fixed.
> 
> Also noticed that the "Copy Link" button does not work for inbox posts, but I'm not sure that it needs to. We only need to have a link for public posts created by our authors, so we could just hide this button for inbox posts. Any thoughts?
> 
> I'll let @tylerelston or @JawadHossain address comments relating to the other PR.

Thanks for the feedback! The issues should be addressed now. Copying links to friends/inbox posts now works, and the post can only be viewed if you log in and are following the author. Links to public posts can be viewable by anyone.

**Comment by JawadHossain** [comment]
@mxiuwang Missing file tagtemplates/shared_post.html

**Comment by mxiuwang** [comment]
> @mxiuwang Missing file tagtemplates/shared_post.html

Oh sorry haha it's there now

File: code/socialDistribution/templates/tagtemplates/shared_post.html [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{% extends 'navBase.html' %}
+
+{% block head %}
+    <title>Author Profile</title>
```
Comment (JawadHossain) [inline code/socialDistribution/templates/tagtemplates/shared_post.html:4]: Maybe call Post View or something similar

**Review by JawadHossain — APPROVED** [review]
Nice, Unlisted public posts are visible when not logged in.

File: code/socialDistribution/templates/tagtemplates/shared_post.html (hunk 3 above)
Comment (mxiuwang) [reply code/socialDistribution/templates/tagtemplates/shared_post.html]: Good catch. Updated.

**Comment by tylerelston** [comment]
Is the copy button in the Inbox working for you? It works well on the homepage/author profile but nothing happens when I click the copy button in the Inbox

**Comment by JawadHossain** [comment]
> Is the copy button in the Inbox working for you? It works well on the homepage/author profile but nothing happens when I click the copy button in the Inbox

@mxiuwang For inbox posts, I think we can hide the copy link button as the user story is only for unlisted posts.
Would be great if you try sharing inbox posts in another pr.

**Comment by mxiuwang** [comment]
> Is the copy button in the Inbox working for you? It works well on the homepage/author profile but nothing happens when I click the copy button in the Inbox


> > Is the copy button in the Inbox working for you? It works well on the homepage/author profile but nothing happens when I click the copy button in the Inbox
> 
> @mxiuwang For inbox posts, I think we can hide the copy link button as the user story is only for unlisted posts. Would be great if you try sharing inbox posts in another pr.

Hid the copy link button on inbox posts

**Comment by debecerra** [comment]
Looks good now, thank you for the fix!

**Review by debecerra — APPROVED** [review]

**Pull request merged**